### PR TITLE
Filter non email owners from labels 

### PR DESCRIPTION
**Pull request by ANich**
In some cases we have owners that are GitHub identities, e.g.: https://github.com/mozilla/bigquery-etl/blob/5b178559b3f7b9de4ed5f12763b685c29d09d67d/sql_generators/active_users/templates/metadata.yaml#L17-L19 

These aren't always valid labels and currently show up as empty strings in BQ, so we should skip them entirely. ~Also we use emails verbatim as owners in dags so it makes sense here too I think.~

### CODE REVIEW

File: bigquery_etl/metadata/parse_metadata.py [hunk 1]
```diff
@@ -256,10 +261,9 @@ def from_file(cls, metadata_file):
                 if "owners" in metadata:
                     owners = metadata["owners"]
                     for i, owner in enumerate(metadata["owners"]):
-                        label = owner.split("@")[0]
-                        if not Metadata.is_valid_label(label):
-                            label = ""
-                        labels[f"owner{i+1}"] = label
+                        if not is_email(owner):
```
Comment (wwyc) [inline bigquery_etl/metadata/parse_metadata.py:264]: I think emails cannot be labels because "@" is not allowed?
https://cloud.google.com/bigquery/docs/labels-intro#requirements

Comment (ANich) [reply bigquery_etl/metadata/parse_metadata.py:264]: Ahh fair enough, I'll update this to just skip non-email labels then

File: bigquery_etl/metadata/parse_metadata.py [hunk 2]
```diff
@@ -166,6 +166,11 @@ def validate_labels(self, attribute, value):
                             raise ValueError(f"Invalid label format: {bug}")
                 else:
                     raise ValueError("Error: review_bugs needs to be a list.")
+            elif re.match(r"owner\d", key):
+                if not is_email(label):
+                    raise ValueError(
```
Comment (lucia-vargas-a) [inline bigquery_etl/metadata/parse_metadata.py:171]: Should we skip Github identities instead of raising an exception? E.g.
if not is_email_or_github_identity(label):

File: tests/metadata/test_parse_metadata.py [hunk 3]
```diff
@@ -84,9 +84,13 @@ def test_from_file(self):
...
-        assert metadata.owners == ["test1@mozilla.com", "test2@example.com"]
-        assert metadata.labels["owner1"] == "test1"
-        assert metadata.labels["owner2"] == "test2"
+        assert metadata.owners == [
+            "test1@mozilla.com",
+            "test2@example.com",
+            "mozilla/test_group",
+        ]
+        assert metadata.labels["owner1"] == "test1@mozilla.com"
+        assert metadata.labels["owner2"] == "test2@example.com"
```
Comment (lucia-vargas-a) [inline tests/metadata/test_parse_metadata.py:93]: Could we add a test to assert the case of owner "mozilla/test_group"? Expects an exception (or None).

**Review by lucia-vargas-a — APPROVED** [review]
